apollo-federation/src/sources/connect/validation/entity.rs

```diff
@@ -64,8 +79,380 @@ pub(super) fn validate_entity_arg(
                         .collect(),
                 })
             }
+
+            // Validate the arguments to the entity resolver (but if the field was determined to be
+            // invalid for an entity resolver above, don't bother validating the field arguments).
+            if messages.is_empty() {
+                if field.arguments.is_empty() {
+                    messages.push(Message {
+                        code: Code::EntityResolverArgumentMismatch,
+                        message: format!(
+                            "{coordinate} must have arguments. See https://preview-docs.apollographql.com/graphos/connectors/directives/#rules-for-entity-true",
+                            coordinate = field_with_connect_directive_entity_true_coordinate(
+                                connect_directive_name,
+                                entity_arg_value.as_ref(),
+                                object,
+                                &field.name,
+                            ),
+                        ),
+                        locations: entity_arg
+                            .line_column_range(source_map)
+                            .into_iter()
+                            .collect(),
+                    });
+                } else if let Some(object_type) = schema.get_object(field.ty.inner_named_type()) {
+                    let key_fields = object_type
+                        .directives
+                        .iter()
+                        .filter(|directive| directive.name == "key")
+                        .filter_map(|directive| {
+                            directive.arguments.iter().find(|arg| arg.name == "fields")
```

Comment:
you could also check the `resolvable` argument — if it's false, we don't care about this `@key`. a non-resolvable key is just a way to keep entity definitions consistent across subgraphs and avoid having to use `@shareable`

Reply:
Done.

Reply:
Do you have an example of how `resolvable` would be used with an entity connector? I'm not understanding how that would impact this validation.

Reply:
Done. A key with `resolvable` set to `false` will now be ignored (but will not cause any errors).